File: occasum-core/src/main/java/pw/stamina/occasum/properties/primitives/clamp/MinDoubleClamp.java
package pw.stamina.occasum.properties.primitives.clamp;

final class MinDoubleClamp implements DoubleClamp {
    private final double min;

    MinDoubleClamp(double min) {
        this.min = min;
    }

    @Override
    public double clamp(double value) {
        if (value < min) {
            return min;
        } else {
            return value;
        }
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/properties/primitives/clamp/MinDoubleClamp.kt
package pw.stamina.occasum.properties.primitives.clamp

internal class MinDoubleClamp(private val min: Double) : DoubleClamp {

    override fun clamp(value: Double): Double {
        return if (value < min) {
            min
        } else {
            value
        }
    }
}
